packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1532,12 +1532,21 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
     if (kFlutterMemoryAllocationsEnabled) {
       ChangeNotifier.maybeDispatchObjectCreation(this);
     }
-    if (kIsWeb || defaultTargetPlatform != TargetPlatform.android) {
+    final bool mobileDevice = switch (defaultTargetPlatform) {
```

Comment:
Consider moving this to a private getter so it can be reused in all places in this file.